### PR TITLE
new formula: oclint

### PR DESCRIPTION
**Pull request by smoofra**
http://oclint.org/

OCLint is a static code analysis tool for improving quality and reducing defects by inspecting C, C++ and Objective-C code and looking for potential problems

### CODE REVIEW

File: oclint.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class Oclint < Formula
+  desc "A clang-based static analyser for C, C++, and Objective C"
+  homepage "http://oclint.org"
+  ### doesn't build, I think because of some clang API change.x
```
Comment (dunn) [inline oclint.rb]: Does this mean that it might soon qualify for being in core?

Comment (smoofra) [inline oclint.rb]: I'm not sure when they're planning on releasing 0.9, but I expect when they do it would qualify.

**Comment by smoofra** [comment]
@DomT4  is that jenkins failure my fault?

**Comment by DomT4** [comment]
Nope. Have asked it to look again.

**Comment by smoofra** [comment]
bump

File: oclint.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
+class Oclint < Formula
+  desc "A clang-based static analyser for C, C++, objC"
+  homepage "http://oclint.org"
+  ### doesn't build, I think because of some clang API change.x
```
Comment (DomT4) [inline oclint.rb]: Let's leave out the `url` and `sha256` and just include the line about `current stable version doesn't build` or such.

File: oclint.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  # url "http://archives.oclint.org/releases/0.8/oclint-0.8.1-src.tar.gz"
+  # sha256 "fb6dab9ac619bacfea42e56469147cfc40e680642cedf352b87986c0bf1f7510"
+
+  head do
+    url "https://github.com/oclint/oclint.git"
+    resource "oclint-xcodebuild" do
```
Comment (DomT4) [inline oclint.rb]: Empty line between the `url` and `resource` please.

File: oclint.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    bin.install "oclint-xcodebuild/oclint-xcodebuild"
+
+    chdir "oclint-scripts" do
+      system "sh", "./makeWithExternClang", "#{HOMEBREW_PREFIX}/opt/llvm"
+    end
+    cp_r Dir["./build/oclint-release/*"], prefix
+  end
+
+  test do
+    mktemp do
```
Comment (DomT4) [inline oclint.rb]: You're already in a temporary directory here. You shouldn't need to create a new one?

File: oclint.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      File.open("foo.c", "w") do |f|
```
Comment (DomT4) [inline oclint.rb]: Could you not just do `(testpath/"foo.c").write <<-EOS.undent` and such?

**Comment by smoofra** [comment]
ok, those issues should be fixed

**Comment by smoofra** [comment]
bump

File: oclint.rb [hunk 6]
```diff
@@ -0,0 +1,45 @@
+class Oclint < Formula
+  desc "A clang-based static analyser for C, C++, objC"
+  homepage "http://oclint.org"
+
+  ### oclint-0.8.1 doesn't build, I think because of some clang API changes.
```
Comment (DomT4) [inline oclint.rb]: Can we report this upstream somewhere so we can track that.

File: oclint.rb [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
```
Comment (DomT4) [inline oclint.rb]: Don't need the empty line here.

File: oclint.rb [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  depends_on "llvm" => "with-clang"
+
+  def install
+    # Homebrew llvm libc++.dylib doesn't correctly reexport libc++abi
+    ENV.append("LDFLAGS", "-lc++abi")
+
+    (buildpath/"oclint-xcodebuild").install resource("oclint-xcodebuild")
+    bin.install "oclint-xcodebuild/oclint-xcodebuild"
+
+    chdir "oclint-scripts" do
```
Comment (DomT4) [inline oclint.rb]: Please use `cd` instead of `chdir`

File: oclint.rb [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+      system "sh", "./makeWithExternClang", "#{HOMEBREW_PREFIX}/opt/llvm"
```
Comment (DomT4) [inline oclint.rb]: `Formula["llvm"].opt_prefix` rather than `#{HOMEBREW_PREFIX}/opt/llvm`

File: oclint.rb [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    end
+    cp_r Dir["./build/oclint-release/*"], prefix
```
Comment (DomT4) [inline oclint.rb]: `prefix.install Dir["./build/oclint-release/*"]`

File: oclint.rb [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+  end
+
+  test do
+    (testpath/"foo.c").write(<<-END.undent)
```
Comment (DomT4) [inline oclint.rb]: Use `EOS` rather than `END`

**Comment by bfontaine** [comment]
Sorry for the late response.
We’re deprecating this repository and won’t accept new formulae. Please ask upstream for a stable release and submit to the [core repository](https://github.com/Homebrew/homebrew-core) or [maintain the formula in a tap](https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/How-to-Create-and-Maintain-a-Tap.md). Thanks!

**Pull request closed without merging**